UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs

```diff
@@ -868,6 +870,46 @@ public void CmdRequestSpell(int spellIndex, Vector3 clickPosition)
 		}
 	}
 
+	[Command]
+	public void CmdRequestChangelingAbilites(int abilityIndex, Vector3 clickPosition)
+	{
+		foreach (var spell in playerScript.Mind.ChangelingAbilities)
+		{
+			if (spell.AbilityData.Index == abilityIndex)
+			{
+				spell.CallActionServer(PlayerList.Instance.GetOnline(gameObject), clickPosition);
```

Comment:
double check that they can't do anything funny with clickPosition , since the client could send anything

Reply:
I tests that inside abilities when calculating distance from changeling to player - target